src/data/sum/order.lean

```diff
@@ -0,0 +1,491 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.hom.basic
+import order.lexicographic
+
+/-!
+# Orders on a sum type
+
+This file defines the disjoint sum and the linear (aka lexicographic) sum of two orders and provides
+relation instances for `sum.lift_rel` and `sum.lex`.
+
+We declare the disjoint sum of orders as the default set of instances. The linear order goes on a
+type synonym.
```

Comment:
I think it would be good to explicitly provide links to the corresponding `has_le` and `has_lt` instances in this docstring.